engines/private/private.cpp

```diff
@@ -0,0 +1,1051 @@
+
+    // Debug channels
+    DebugMan.addDebugChannel(kPrivateDebugFunction, "functions", "Function execution debug channel");
+    DebugMan.addDebugChannel(kPrivateDebugCode, "code", "Code execution debug channel");
+    DebugMan.addDebugChannel(kPrivateDebugScript, "script", "Script execution debug channel");
+
+    // Global object for external reference
+    g_private = this;
+
+    // Setting execution
+    _nextSetting = NULL;
+    _currentSetting = NULL;
+    _pausedSetting = NULL;
+    _modified = false;
+    _mode = -1;
+    _toTake = false;
+
+    // Movies
+    _nextMovie = NULL;
+    _nextVS = NULL;
+    _repeatedMovieExit = new Common::String("");
+
+    // Save and load
+    _saveGameMask = NULL;
+    _loadGameMask = NULL;
+
+    // Interface
+    _framePath = new Common::String("inface/general/inface2.bmp");
+
+    // Police
+    _policeBustEnabled = false;
+    _policeBustSetting = NULL;
+    _numberClicks = 0;
+    policeVideoIndex = 0;
+    _sirenSound = Common::String("po/audio/posfx002.wav");
+
+    // General sounds
+    _globalAudioPath = Common::String("global/audio/");
+    _noStopSounds = false;
+
+    // Radios and phone
+    _policeRadioArea = NULL;
+    _AMRadioArea = NULL;
+    _phoneArea = NULL;
+    // TODO: use this as a default sound for radio
+    _infaceRadioPath = Common::String("inface/radio/");
+    _phonePrefix = Common::String("inface/telephon/");
+    _phoneCallSound = Common::String("phone.wav");
+
+    // Dossiers
+    _dossierPage = 0;
+    _dossierSuspect = 0;
+    _dossierNextSuspectMask = NULL;
+    _dossierPrevSuspectMask = NULL;
+
+    // Diary
+    _diaryLocPrefix = Common::String("inface/diary/loclist/");
+}
+
+PrivateEngine::~PrivateEngine() {
+    // Dispose your resources here
+    delete _rnd;
+
+    // Remove all of our debug levels
+    DebugMan.clearAllDebugChannels();
+}
+
+void PrivateEngine::initializePath(const Common::FSNode &gamePath) {
+    SearchMan.addDirectory(gamePath.getPath(), gamePath, 0, 10);
+}
+
+void PrivateEngine::setOrigin(const int point[2]) {
+    delete _origin;
+    _origin = new Common::Point(point[0], point[1]);;
+}
+
+Common::Error PrivateEngine::run() {
+
+    assert(_installerArchive.open("SUPPORT/ASSETS.Z"));
+    Common::SeekableReadStream *file = NULL;
+    // if the full game is used
+    if (!isDemo()) {
+        assert(_installerArchive.hasFile("GAME.DAT"));
+        file = _installerArchive.createReadStreamForMember("GAME.DAT");
+    } else {
+        // if the demo from archive.org is used
+        if (_installerArchive.hasFile("GAME.TXT"))
+            file = _installerArchive.createReadStreamForMember("GAME.TXT");
+
+        // if the demo from the full retail CDROM is used
+        else {
+            if (_installerArchive.hasFile("DEMOGAME.DAT"))
+                file = _installerArchive.createReadStreamForMember("DEMOGAME.DAT");
+        }
+    }
+
+    // Read assets file
+    assert(file != NULL);
+    char *buf = (char *)malloc(file->size()+1);
```

Comment:
If you think that the international releases may have characters with diacritics, e.g. with code >0x7f, I'd recommend using `byte` here.

Reply:
There is no text rendering and the european releases won't work unless the bytecode interpreter is implemented.